app/controllers/scenarios_controller.rb

```diff
@@ -95,6 +95,16 @@ def update
     end
   end
 
+  def enable_or_disable_all_agents
+    @scenario = current_user.scenarios.find(params[:id])
+    @scenario.agents.update_all disabled: params[:scenario][:disabled]
+
+    respond_to do |format|
+      format.html { redirect_to @scenario }
+      flash[:notice] = 'The agents has been successfully updated.'
+    end
+  end
+
```

Comment:
I whittled this down to a bit because of rubocop assignment branch error (method does too much). I think this is more akin to the `destroy` method below it than the `put` method above it. The likelihood that this will fail is on the low side.